### ISSUE TITLE
Why nn.BCECriterion perform size averaging by nElement instead of total weights?

### ISSUE THREAD

**Issue by AlexBYin** [torch/nn#926]
In nn.BCECriterion, both the output loss and gradients are normalized by N - the number of elements in the input vector. 
But if weights ~= nil thus loss and gradients are weighted, shouldn't them be normalized by torch.sum(self.weights)? 
Any particular reason why we are not using the total_weights for size averaging?
